### PR TITLE
Add leo27945875 in hw2

### PR DESCRIPTION
**Pull request by leo27945875**
Please merge the changes of leo27945875 hw2.

---
A good PR should include the following parts:

- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Make sure tests/CI pass

### CODE REVIEW

File: hw2/leo27945875/q2/test_q2.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import _vector
+import math
+
+
+def test_calc_angle():
+    angle = _vector.calc_angle(0., 1., 1., 0.)
+    assert abs(angle - 0.5 * math.pi) < 1e-5
```
Comment (ThreeMonth03) [inline hw2/leo27945875/q2/test_q2.py]: Could you please add the sufficient unittest according to the `validate.sh`?

https://github.com/yungyuc/nsdhw_24au/blob/3d417a41e674361630b8994f220952283ab046e9/hw2/validate.sh#L131-L138

**Review by ThreeMonth03 — CHANGES_REQUESTED** [review]
- [ ] Add the sufficient testcase in Q2.

File: hw2/leo27945875/q2/test_q2.py (hunk 1 above)
Comment (leo27945875) [reply hw2/leo27945875/q2/test_q2.py]: Hi, I have added more testcases.

**Review by ThreeMonth03 — APPROVED** [review]
Thanks, now you get the 5 points in this homework.

**Pull request merged**